src/factory.ts

```diff
@@ -28,26 +28,98 @@ import {DEFAULT_LABELS} from './constants';
 // eslint-disable-next-line @typescript-eslint/no-var-requires
 const parseGithubRepoUrl = require('parse-github-repo-url');
 
+// types defining methods available to call on instances
+export type ReleasePRMethod = 'run' | 'latestTag';
+export type GitHubReleaseMethod = 'run';
+export type Method = ReleasePRMethod | GitHubReleaseMethod;
+
+// types defining cli commands and their options
+export type ReleasePRCommands = 'release-pr' | 'latest-tag';
+export type GitHubReleaseCommands = 'github-release';
+type Command = ReleasePRCommands | GitHubReleaseCommands;
+type IsReleasePRCmd = {
+  command: Command;
+  options: ReleasePRFactoryOptions;
+};
+type IsGitHubReleaseCmd = {
+  command: Command;
+  options: GitHubReleaseFactoryOptions;
+};
+
+// shorthand aliases for instance/method call return types
+export type ReleasePRRunResult = Promise<number | GitHubTag | undefined>;
+export type GitHubReleaseRunResult = Promise<GitHubReleaseResponse | undefined>;
+export type RunResult = ReleasePRRunResult | GitHubReleaseRunResult;
```

Comment:
a bit of an explosion of types here... I'm definitely open to a simpler approach. But this is where I got adding support for `latest-tag` while keeping the nice/testable pattern of

CLI <-> translate to instance/method (`runCommand()`) <-> execute (`call()`)

I could see other CLI commands being added too (maybe `validate/list-package-config` for the aggregate) so I think it's a worthwhile endeavor.